include/json/reader.h

```diff
@@ -271,6 +271,7 @@ class JSON_API CharReader {
      * \throw std::exception if something goes wrong (e.g. invalid settings)
      */
     virtual CharReader* newCharReader() const = 0;
+    virtual std::unique_ptr<CharReader> makeCharReader() const = 0;
```

Comment:
Not virtual. That's not backward compatible or necessary.

Reply:
This is an Interface, all the functions should be virtual, or am I missing something?

Reply:
I don't know why you think that.

Reply:
Line 243 there is a remark 
/** Interface for reading JSON from a char array.
 */

Reply:
Yes sorry. I meant that I understand that it's an interface I don't know why that means everything has to be virtual

Reply:
in case of that code 
CharReader::Factory* ptr = new CharReaderBuilder();
    ptr->makeCharReader();
we want to invoke CharReaderBuilder::makeCharReader() and not the interface class

Reply:
the other possibility is to make the implementation on the Interface class, and remove the virtual.

Reply:
done